controllers/secretagentconfiguration_controller.go

```diff
@@ -78,10 +73,30 @@ func (reconciler *SecretAgentConfigurationReconciler) Reconcile(req ctrl.Request
 		log.Error(err, "unable to fetch SecretAgentConfiguration")
 		return ctrl.Result{}, err
 	}
-	// Stop watching for secret changes while the reconcile loop is running
-	watchOwnedObjects = false
-	defer func() { watchOwnedObjects = true }()
 	log.V(1).Info("** Reconcile loop start **")
+	saFinalizerName := "secret-agent.secrets.forgerock.io/" + instance.Namespace + "." + instance.Name
+
+	// check DeletionTimestamp to determine if the SAC is under deletion
+	if instance.ObjectMeta.DeletionTimestamp.IsZero() {
```

Comment:
I think this would read better if it were `if ! instance.ObjectMeta.DeletionTimestamp.IsZero()` then there would be no else required.